src/components/AccordionPanel/AccordionPanel.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useState } from 'react'
+
+import { mapToArray, arrayToMap } from './AccordionPanelHelper'
+
+type Props = {
+  children: React.ReactNode
+  icon?: 'left' | 'right' | 'none'
```

Comment:
[IMO] The name of the props `icon` is not intuitive. Whether or not the icon is displayed and the display position of the icon are different interests. It is difficult to guess 'left' | 'right' | 'none' from the name `icon`.
There should be boolean props to display the icon and 'left' | 'right' optional props to determine the position of the icon.